android/capacitor/src/main/java/com/getcapacitor/plugin/LocalNotification.java

```diff
@@ -0,0 +1,199 @@
+    super.load();
+    createDefaultNotificationChannel();
+  }
+
+  private void createDefaultNotificationChannel() {
+    // Create the NotificationChannel, but only on API 26+ because
+    // the NotificationChannel class is new and not in the support library
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+      CharSequence name = "Default";
+      String description = "Default";
+      int importance = NotificationManager.IMPORTANCE_DEFAULT;
+      NotificationChannel channel = new NotificationChannel(CHANNEL_ID, name, importance);
+      channel.setDescription(description);
+      // Register the channel with the system; you can't change the importance
+      // or other notification behaviors after this
+      NotificationManager notificationManager = getContext().getSystemService(NotificationManager.class);
+      notificationManager.createNotificationChannel(channel);
+    }
+  }
+
+  // TODO unify input parameters for both IOS and Android
+
+  /**
+   * Schedule a notification call from JavaScript
+   * Creates local notification in system.
+   */
+  @PluginMethod()
+  public void schedule(PluginCall call) {
+    List<Notification> notifications = this.buildNotificationList(call);
+    if (notifications == null) {
+      return;
+    }
+    JSONArray ids = new JSONArray();
+    NotificationManagerCompat notificationManager = NotificationManagerCompat.from(this.getContext());
+    boolean notificationsEnabled = notificationManager.areNotificationsEnabled();
+    if (!notificationsEnabled) {
+      call.error("Notifications not enabled on this device");
+      return;
+    }
+    for (Notification notification : notifications) {
+      if (notification.getId() == null) {
+        call.error("Notification missing identifier");
+        return;
+      }
+
+      Intent intent = new Intent(getContext(), getActivity().getClass());
+      intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+      PendingIntent pendingIntent = PendingIntent.getActivity(getContext(), 0, intent, 0);
+
+      NotificationCompat.Builder mBuilder = new NotificationCompat.Builder(this.getContext(), CHANNEL_ID)
+              .setContentTitle(notification.getTitle())
+              .setContentText(notification.getBody())
+              .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+              .setContentIntent(pendingIntent)
+              .setAutoCancel(true);
+
+
+      String sound = notification.getSound();
+      if (sound != null) {
+        mBuilder.setSound(Uri.parse(sound));
+
+      }
+
+      JSONObject extra = notification.getExtra();
+      if (extra != null) {
+        Bundle extras = new Bundle();
+        extras.putString("data", extra.toString());
+        mBuilder.addExtras(extras);
+        intent.putExtras(extras);
+      }
+
+      // notificationId is a unique int for each notification that you must define
+      notificationManager.notify(Integer.valueOf(notification.getId()), mBuilder.build());
```

Comment:
Interesting, will change the iOS implementation and example code to use ints too

Reply:
Going to do that in this PR.

Reply:
Too late, is already changed 😜